master/internal/authz/permission_denied_error.go

```diff
@@ -0,0 +1,28 @@
+package authz
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/determined-ai/determined/proto/pkg/rbacv1"
+)
+
+// PermissionDeniedError represents an error that arises when a user does not have sufficient
+// access privileges. RequiredPermissions can be empty for non-rbac errors.
+type PermissionDeniedError struct {
+	RequiredPermissions []*rbacv1.Permission
+}
+
+// Error returns an error string.
+func (p PermissionDeniedError) Error() string {
+	if len(p.RequiredPermissions) == 0 {
+		return "access denied"
```

Comment:
for OSS stuff it would be cool if we could pass in a specific message

It would feel frustrating to just get access denied and not really sure what went wrong where

Reply:
Maybe it's best if we had a meeting together tomorrow in the AM to discuss more, but I did talk about this also with Max, and adding a custom message could be subverting the purpose of the `mapAndFilterErrors` function, which was to only let certain errors through.